pinot-core/src/main/java/com/linkedin/pinot/core/plan/TransformPlanNode.java

```
@@ -0,0 +1,172 @@
+        expressionTrees.add(expressionTree);
+      }
+    }
+    return expressionTrees;
+  }
+
+  /**
+   * Helper method to extract projection columns and transform expressions from the given
+   * BrokerRequest.
+   *  @param brokerRequest BrokerRequest to process
+   * @param projectionColumns Output projection columns from broker request
+   * @param transformExpressions Output transform expression from broker request
+   */
+  private void extractColumnsAndTransforms(BrokerRequest brokerRequest, Set<String> projectionColumns,
+      Set<String> transformExpressions) {
+
+    if (brokerRequest.isSetAggregationsInfo()) {
+      // TODO: Add transform support.
+      for (AggregationInfo aggregationInfo : brokerRequest.getAggregationsInfo()) {
+        if (!aggregationInfo.getAggregationType()
+            .equalsIgnoreCase(AggregationFunctionFactory.AggregationFunctionType.COUNT.getName())) {
+          String columns = aggregationInfo.getAggregationParams().get("column").trim();
+          projectionColumns.addAll(Arrays.asList(columns.split(",")));
+        }
+      }
+
+      // Collect all group by related columns.
+      if (brokerRequest.isSetGroupBy()) {
+        GroupBy groupBy = brokerRequest.getGroupBy();
+        List<String> groupByColumns = groupBy.getColumns();
+
+        // GroupByColumns can be null if all group-bys are transforms.
+        if (groupByColumns != null) {
+          projectionColumns.addAll(groupByColumns);
+        }
+
+        // Check null for backward compatibility.
+        List<String> expressions = groupBy.getExpressions();
+        if (expressions != null) {
+          transformExpressions.addAll(expressions);
+          if (groupByColumns != null && !groupByColumns.isEmpty()) {
+            transformExpressions.removeAll(groupByColumns);
+          }
+        }
+      }
+    } else {
+      // TODO: Add transform support.
+      projectionColumns.addAll(brokerRequest.getSelections().getSelectionColumns());
```

Selection query should never reach here, and it will break if it reached here. So please throw an exception for now.